openeogeotrellis/opensearch.py

```diff
@@ -22,11 +22,29 @@ def __init__(self, endpoint: str):
 
     def _get_collection(self, collection_id: str) -> dict:
         if not self._cache:
-            url = self.endpoint + "/collections"
-            logger.info(f"Getting collection metadata from {url}")
-            resp = requests.get(url=url)
-            resp.raise_for_status()
-            self._cache = {f["id"]: f for f in resp.json()["features"]}
+            cache = {}
+            start_index = 1
+            while True:
+                url = f"{self.endpoint}/collections?startIndex={start_index}&count=200"
+                logger.info(f"Getting collection metadata from {url}")
+                resp = requests.get(url=url)
+                resp.raise_for_status()
+                json = resp.json()
+                features = json["features"]
+                cache_length_before = len(cache)
+                for f in features:
+                    cache[f["id"]] = f
+                if cache_length_before == len(cache):
```

Comment:
Clarify that this means: if no new items were added to the cache...

Or at least reverse the comparison because this is confusing.